zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java

```diff
@@ -670,6 +675,48 @@ public void shouldApplyOutputMappings() {
         .contains(tuple("adHocResult", "[1,2]", processInstanceKey));
   }
 
+  @Test
+  public void shouldCreateLocalAdHocSubProcessElementsVariable() {
+    // given
+    final BpmnModelInstance process =
+        process(
+            adHocSubProcess -> {
+              adHocSubProcess.task("A");
+            });
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariable("activateElements", List.of("A"))
+            .create();
+
+    final var adHocSubProcessKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(AD_HOC_SUB_PROCESS_ELEMENT_ID)
+            .getFirst()
+            .getKey();
+
+    // then
+    // actual variable assertions are done in detail in AdHocSubProcessElementsVariableTest
```

Comment:
This test is kind of redundant with the variable test, but the idea was to have at least one high-level test in here as well. Happy to remove if we don't see any value.